ublox/src/ubx_packets/packets.rs

```diff
@@ -2204,6 +2206,71 @@ struct MgaGpsEPH {
     reserved3: [u8; 2],
 }
 
+#[ubx_packet_recv]
+#[ubx(class = 0x02, id = 0x15, max_payload_len = 1240)]
+struct RxmRawx {
+    rcv_tow: [u8; 8],
```

Comment:
these are f64s, will have to manually convert / support conversion somehow
using `[ubx(map_type = f64, from = ...)]` with custom fn causes
```
2224 | #[ubx_packet_recv]
     | ^^^^^^^^^^^^^^^^^^
     | |
     | expected `&[u8]`, found array `[u8; 8]`
     | expected `&[u8]` because of return type
```

Reply:
besides that everything works as expected

Reply:
Not sure it's the best way to go about it but this works I think

```rust
#[ubx_packet_recv]
#[ubx(class = 0x02, id = 0x15, max_payload_len = 1240)]
struct RxmRawx {
    #[ubx(
        map_type = f64,
        from = convert_to_f64_unchecked,
        is_valid = is_f64_valid,
        may_fail,
        get_as_ref
    )]
    rcv_tow: [u8; 8],
    week: u16,
    leap_s: i8,
    num_meas: u8,
    rec_stat: u8,
    reserved1: [u8; 3],
    #[ubx(map_type = RxmRawxInfoIter,
    from = RxmRawxInfoIter::new,
    is_valid = RxmRawxInfoIter::is_valid)]
    iter: [u8; 0],
}

fn convert_to_f64_unchecked(bytes: &[u8]) -> f64 {
    let bytes: [u8; 8] = bytes[..8].try_into().unwrap();
    f64::from_le_bytes(bytes)
}

fn is_f64_valid(bytes: &[u8]) -> bool {
    bytes.len() >= 8
}
```

Reply:
Actually doing this

```
diff --git a/ublox_derive/src/input.rs b/ublox_derive/src/input.rs
index acee075..0ce2855 100644
--- a/ublox_derive/src/input.rs
+++ b/ublox_derive/src/input.rs
@@ -549,13 +549,14 @@ impl Parse for Comment {
 fn field_size_bytes(ty: &Type) -> syn::Result<Option<NonZeroUsize>> {
     //TODO: make this array static
     //TODO: support f32, f64
-    let valid_types: [(Type, NonZeroUsize); 6] = [
+    let valid_types: [(Type, NonZeroUsize); 7] = [
         (syn::parse_quote!(u8), NonZeroUsize::new(1).unwrap()),
         (syn::parse_quote!(i8), NonZeroUsize::new(1).unwrap()),
         (syn::parse_quote!(u16), NonZeroUsize::new(2).unwrap()),
         (syn::parse_quote!(i16), NonZeroUsize::new(2).unwrap()),
         (syn::parse_quote!(u32), NonZeroUsize::new(4).unwrap()),
         (syn::parse_quote!(i32), NonZeroUsize::new(4).unwrap()),
+        (syn::parse_quote!(f64), NonZeroUsize::new(8).unwrap()),
     ];
     if let Some((_ty, size)) = valid_types.iter().find(|x| x.0 == *ty) {
         Ok(Some(*size))

```

and then just `rec_tow: 64` also seems to work. not sure why it was excluded tho so might cause an issue?

Reply:
same also for f32, not sure what `//TODO: support f32, f64` is about

Reply:
oh bruh i tried that before and was giving me errors so thought we couldnt do that but now it works somehow

Reply:
I assume there was some reason the original author didn't do this, so it might break something. Just keep that in mind and then the other way can be a fallback. Or maybe the original author just forgot or something lol